curp/src/server/raw_curp/mod.rs

```diff
@@ -776,6 +784,57 @@ impl<C: 'static + Command, RC: RoleChange + 'static> RawCurp<C, RC> {
             ReadState::Ids(IdSet::new(ids))
         }
     }
+
+    /// Handle `move_leader`
+    pub(super) fn handle_move_leader(&self, node_id: ServerId) -> Result<bool, CurpError> {
+        debug!("{} received move leader", self.id());
+        let st_r = self.st.read();
+        if st_r.role != Role::Leader {
+            return Err(CurpError::Redirect(st_r.leader_id, st_r.term));
+        }
+        if !self.cluster().get(&node_id).is_some_and(|m| !m.is_learner) {
+            return Err(CurpError::LeaderTransfer(
```

Comment:
Can we make a learning be a leader in ETCD?

Reply:
Learner cannot be a leader directly in both Xline and Etcd.